cpp/src/c_api/graph_sg.cpp

```diff
@@ -173,26 +175,28 @@ struct destroy_graph_functor : public abstract_functor {
 extern "C" cugraph_error_code_t cugraph_sg_graph_create(
   const cugraph_resource_handle_t* handle,
   const cugraph_graph_properties_t* properties,
-  const cugraph_type_erased_device_array_t* src,
-  const cugraph_type_erased_device_array_t* dst,
-  const cugraph_type_erased_device_array_t* weights,
+  const cugraph_type_erased_device_array_view_t* src,
+  const cugraph_type_erased_device_array_view_t* dst,
+  const cugraph_type_erased_device_array_view_t* weights,
   bool_t store_transposed,
   bool_t renumber,
   bool_t check,
   cugraph_graph_t** graph,
   cugraph_error_t** error)
 {
   constexpr bool multi_gpu = false;
-  constexpr size_t int32_threshold{2 ^ 31 - 1};
+  constexpr size_t int32_threshold{std::numeric_limits<int32_t>::max()};
```

Comment:
Why int32 here? Are we assuming 32 bit vertex IDs? (for SG) This might be OK for now (may be not if managed memory) but doesn't sound future-proof.

Reply:
This threshold is identifying whether to create a 32 or 64-bit edge_t.  If the number of edges is less than the threshold then we will use edge_t as int32_t, otherwise we will use int64_t.  The constant is used further down in the code to make this choice.

This does assume an int32_t vertex_t.  I suppose we could have the case where the vertex_t is int64_t but the number of edges is less than 2^31-1.  The if statement below could be modified to catch this case.  @rlratzel can we add a FIXME for this?

Reply:
added FIXME in the next commit.